msupply-foundation/notify
# 228 save disabled popup

Closes #228 

# 👩🏻‍💻 What does this PR do?
 

If a notification has a valid configuration, but the status is set to disabled, show this popup when they hit save:
![Screenshot 2023-11-27 at 5 00 39 PM](https://github.com/msupply-foundation/notify/assets/55115239/baeb7a5b-91f6-4621-9df9-259124156ffb)


# 🧪 How has/should this change been tested?


- Create a new notification
- Fill out required fields, keep the `Enabled` toggle off
- Hit save, this modal should appear
- Select Cancel
- Enable the notification, hit save again - see that saving works right away
- Disable the notification again, and hit save again
- You should get the modal again, this time select Ok
- Your disabled config is saved

## 💌 Any notes for the reviewer?


- I think the wording could be clearer? Of what happens on cancel and ok?
- Wanted a popup with `Enable & Save` and `Save anyway` or something like that, but was limited by the confirmation modal to `Cancel` and `Ok` (if we think the former options are really worth it we could create a new modal, but would take longer...
- Feels a bit clunky that when you hit ok, the notification is saved, but you stay on the same page. For some reason, now that there is a modal, I'm expecting to be redirected back to the list of notification. Does it feel the same to you? Could redirect after... or maybe a success snack popup would help?
